src/main/java/com/tjtanjin/steve/parser/Parser.java

```diff
@@ -0,0 +1,229 @@
+package com.tjtanjin.steve.parser;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+
+import com.tjtanjin.steve.commands.ByeCommand;
+import com.tjtanjin.steve.commands.CommandHandler;
+import com.tjtanjin.steve.commands.DeadlineCommand;
+import com.tjtanjin.steve.commands.DeleteCommand;
+import com.tjtanjin.steve.commands.DoneCommand;
+import com.tjtanjin.steve.commands.EventCommand;
+import com.tjtanjin.steve.commands.FindCommand;
+import com.tjtanjin.steve.commands.HelpCommand;
+import com.tjtanjin.steve.commands.ListCommand;
+import com.tjtanjin.steve.commands.TodoCommand;
+
+/**
+ * Parser class parses user input and returns a response. It also deals directly with the CommandHandler
+ * for command execution.
+ */
+public class Parser {
+
+    //tracks all valid commands
+    private enum Cmd {
+        BYE,
+        LIST,
+        DONE,
+        TODO,
+        DEADLINE,
+        EVENT,
+        DELETE,
+        HELP,
+        FIND
+    }
+
+    //all commands
+    private final ByeCommand BYE_COMMAND;
+    private final DeadlineCommand DEADLINE_COMMAND;
+    private final DeleteCommand DELETE_COMMAND;
+    private final DoneCommand DONE_COMMAND;
+    private final EventCommand EVENT_COMMAND;
+    private final FindCommand FIND_COMMAND;
+    private final HelpCommand HELP_COMMAND;
+    private final ListCommand LIST_COMMAND;
+    private final TodoCommand TODO_COMMAND;
+
+    /**
+     * Constructor for Parser class that initialises all valid commands.
+     */
+    public Parser(CommandHandler commandHandler) {
+        this.BYE_COMMAND = commandHandler.getByeCommand();
+        this.DEADLINE_COMMAND = commandHandler.getDeadlineCommand();
+        this.DELETE_COMMAND = commandHandler.getDeleteCommand();
+        this.DONE_COMMAND = commandHandler.getDoneCommand();
+        this.EVENT_COMMAND = commandHandler.getEventCommand();
+        this.FIND_COMMAND = commandHandler.getFindCommand();
+        this.HELP_COMMAND = commandHandler.getHelpCommand();
+        this.LIST_COMMAND = commandHandler.getListCommand();
+        this.TODO_COMMAND = commandHandler.getTodoCommand();
+    }
+
+    /**
+     * Parses input from user to determine action to take.
+     *
+     * @param input input provided by user
+     * @return string response after operation is done
+     */
+    public String parseInput(String input) {
+
+        String[] parsedInput = input.split(" ", 2);
```

Comment:
parsedInputs?

Reply:
Hello, thank you for raising this up! To clarify, I had actually opted to use a singular form in `parsedInput` to indicate that what was parsed was the original `input` argument. Would this sound appropriate or intuitive?

Reply:
Hi, I agree with TanJin and I think that parsedInput makes sense because the program is handling one line of user input as once.